### PR TITLE
ci(test): test parallel CT runs on self-hosted runners

### PR DESCRIPTION
**Pull request by Traphalet**

### CODE REVIEW

File: .github/workflows/run_test_cases.yaml [hunk 1]
```diff
@@ -10,26 +10,71 @@ on:
   pull_request:
 
 jobs:
-    run_proper_test:
+    prepare:
```
Comment (id) [inline .github/workflows/run_test_cases.yaml:13]: seems like there is inconsistent use of indent=2 and indent=4

File: .github/workflows/run_test_cases.yaml [hunk 2]
```diff
@@ -10,26 +10,71 @@ on:
...
         runs-on: ubuntu-20.04
         container: emqx/build-env:erl23.3.4.9-3-ubuntu20.04
+        outputs:
+          ct_apps: ${{ steps.run_find_apps.outputs.ct_apps }}
+        steps:
+          - uses: actions/checkout@v2
```
Comment (id) [inline .github/workflows/run_test_cases.yaml:19]: please use latest versions of actions when possible

File: .github/workflows/run_test_cases.yaml [hunk 3]
```diff
@@ -10,26 +10,71 @@ on:
...
+              path: source
+              fetch-depth: 0
+          - name: find_ct_apps
+            working-directory: source
+            id: run_find_apps
+            # emqx_plugin_libs doesn't have a test suite -> excluded from app list
+            run: |
+              ct_apps="$(./scripts/find-apps.sh --json | jq -c 'del (.[] | select (. == "apps/emqx_plugin_libs"))')"
+              echo "ct-apps: $ct_apps"
+              echo "::set-output name=ct_apps::$ct_apps"
```
Comment (id) [inline .github/workflows/run_test_cases.yaml:30]: I know we still use this syntax everywhere, but github is already issuing warnings https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/run_test_cases.yaml [hunk 4]
```diff
@@ -10,26 +10,71 @@ on:
...
+          - name: get_all_deps
+            working-directory: source
+            run: |
+              make deps-all
```
Comment (id) [inline .github/workflows/run_test_cases.yaml:34]: this one requires git-credential setup for enterprise

**Review by id — COMMENTED** [review]
some nitpicking so far, looking at test failures

File: .github/workflows/run_test_cases.yaml (hunk 2 above)
Comment (Traphalet) [reply .github/workflows/run_test_cases.yaml]: all actions are now set to currently latest

File: .github/workflows/run_test_cases.yaml [hunk 5]
```diff
@@ -10,140 +10,258 @@ on:
...
+            sleep 5
+          done
+      - name: run common test
+        run: docker exec -i erlang bash -c "make ${{ matrix.app_name }}-ct-pipeline"
+      - name: run cover
+        run: |
+          printenv > .env
+          docker exec -i erlang bash -c "git config --global --add safe.directory /emqx"
+          docker exec -i erlang bash -c "make cover"
+          docker exec --env-file .env -i erlang bash -c "make coveralls"
```
Comment (zmstone) [inline .github/workflows/run_test_cases.yaml:204]: the cover data is uploaded in the step below:
```
      - uses: actions/upload-artifact@v3
        with:
          name: cover
          path: source/_build/test/cover
          if-no-files-found: warn
```
so there is no need to run `make cover` or `make coveralls` here.
```suggestion
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**